src/comps/qb.cc

```diff
@@ -76,6 +81,7 @@ void QB<T>::QB2(
         block_sz = std::min(block_sz, k - curr_sz);
         int next_sz = curr_sz + block_sz;
 
+        ++QB::RF_Obj.RS_Obj.seed;
```

Comment:
This probably won't be sufficient. It should be incremented by [number of generated numbers] after each call to the counter-based RNG.